msal/src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs

```diff
@@ -84,22 +87,20 @@ internal class AuthenticationRequestParameters
                 }
                 else
                 {
-                    if (ClientCredential.Assertion == null || ClientCredential.ValidTo != 0)
+                    if (ClientCredential.Assertion == null || ClientCredential.ValidTo != 0 || ClientCredential.ContainsX5C != SendCertificate)
                     {
-                        bool assertionNearExpiry = (ClientCredential.ValidTo <=
-                                                    Jwt.JsonWebToken.ConvertToTimeT(DateTime.UtcNow +
-                                                                                    TimeSpan.FromMinutes(
-                                                                                    Constants 
-                                                                                    .ExpirationMarginInMinutes)));
-                        if (assertionNearExpiry)
+                        bool assertionExpired = !this.Equals(ClientCredential);
```

Comment:
>!this.Equals(ClientCredential); [](start = 48, length = 31)

You will need a utest to validate that even two different objects with the same assertions will result in being equal. The test may already be there, thus if you can point me to it that would be great.

Reply:
#closed